src/main/java/com/github/stefanbirkner/fakesftpserver/rule/FakeSftpServerRule.java

```diff
@@ -111,17 +111,35 @@
  * <p>The method returns {@code true} iff the file exists and it is not a directory.
  */
 public class FakeSftpServerRule implements TestRule {
-    private static final int PORT = 23454;
+    private static final int DEFAULT_PORT = 23454;
 
     private FileSystem fileSystem;
+    private int port;
+    private int assignedPort;
+
+    public FakeSftpServerRule() {
+        this.port = DEFAULT_PORT;
+    }
+
+    /**
+     * Change the port of the SFTP server.
+     *
+     * @param port Port of the SFTP server
+     * @return This SFTP Server instance
+     */
+    public FakeSftpServerRule withPort(int port) {
+        assert port >= 0;
```

Comment:
Please throw an `IllegalArgumentException` in that case. I don't like `assert` because they have to be activiated explicitly.